packages/flutter/lib/src/widgets/localized_resources.dart

```diff
@@ -0,0 +1,354 @@
+/// classes that contain `Intl.message()` calls. In this case that would just be the
+/// `MyLocalizedResources` class.
+///
+/// One could choose another approach for loading localized resources and looking them up while
+/// still conforming to the structure of this example.
+class LocalizedResources extends StatefulWidget {
+  LocalizedResources({
+    Key key,
+    @required this.locale,
+    this.delegate,
+    this.child
+  }) : super(key: key) {
+    assert(locale != null);
+  }
+
+  /// The resources returned by [LocalizedResources.of] will be specific to this locale.
+  final Locale locale;
+
+  /// This delegate is responsible for loading and caching resources for `locale`.
+  ///
+  /// Typically defined with [DefaultLocalizedResourcesDelegate].
+  final LocalizedResourcesDelegate delegate;
+
+  /// The widget below this widget in the tree.
+  final Widget child;
+
+  /// Returns the localized resources for the widget tree that corresponds to
+  /// [BuildContext] `context`.
+  ///
+  /// This method is typically combined with [LocalizedResourcesData.for] to defined
+  /// a static method that looks up a collection of related related resources. For
+  /// example Flutter looks up Material resources with a method defined like this:
+  ///
+  /// ```dart
+  /// static LocalizedMaterialResources of(BuildContext context) {
+  ///  return LocalizedResources.of(context)
+  ///    .resourcesFor<LocalizedMaterialResources>(LocalizedMaterialResources);
+  /// }
+  /// ```
+  static LocalizedResourcesData of(BuildContext context) {
+    final _LocalizedResourcesScope scope = context.inheritFromWidgetOfExactType(_LocalizedResourcesScope);
+    return scope?.localizedResourcesState;
+  }
+
+  @override
+  _LocalizedResourcesState createState() => new _LocalizedResourcesState();
+}
+
+class _LocalizedResourcesState extends State<LocalizedResources> implements LocalizedResourcesData {
+  final GlobalKey _localizedResourcesScopeKey = new GlobalKey();
+
+  @override
+  Locale get locale => _locale;
+  Locale _locale;
+
+  @override
+  void initState() {
+    super.initState();
+    load(widget.locale);
+  }
+
+  void load(Locale locale) {
+    if (widget.delegate == null) {
+      setState(() {
+        _locale = locale;
+      });
+      return;
+    }
+
+    widget.delegate.load(locale).then((_) {
+      setState(() {
+        _locale = locale;
+      });
+      final InheritedElement scopeElement = _localizedResourcesScopeKey.currentContext;
+      scopeElement?.dispatchDidChangeDependencies();
+    });
+  }
+
+  @override
+  T resourcesFor<T>(Type type) => widget.delegate.resourcesFor<T>(_locale, type);
+
+  @override
+  Widget build(BuildContext context) {
+    return new _LocalizedResourcesScope(
+      key: _localizedResourcesScopeKey,
+      locale: widget.locale,
+      localizedResourcesState: this,
+      child: _locale != null ? widget.child : new Container(),
```

Comment:
The underlying assumption here is that we're going to show the user a single blank frame on startup because localized resource loading is based on Futures. 

For apps with no localizations, apps that load localized resources synchronously (Flutter's TBD localized resources are loaded synchronously), apps that load resources asynchronously but finish during a warm-up frame (while we're waiting for the engine), there will be no blank frame. 

We could allow one to provide a fallback widget here, instead of just building a Container, but that introduces another problem:

Showing a non-blank fallback widget while we wait for the one or two frames that loading needs would force us to show the fallback for many more frames while an animated transition from the fallback content to the real content occurs. If the fallback is only shown for a few frames and then replaced by the real content the app will be appear to flash.

In cases where resource loading is simple but wrapped up in Futures, like a conventional application of the Dart intl package, showing one blank frame - in the event that there isn't a warm-up frame - may be less distracting than showing an animated transition from a fallback widget to the app's content for 100+ ms.